### PR TITLE
GameCommand: Vehicle Rearrange

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Another command.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Seems to be missing an error title.

Screenshot 2023-12-04 at 23 24 49

File: src/OpenLoco/src/GameCommands/Vehicles/VehicleRearrange.cpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    // 0x004AF4D6
+    static void sub_4AF4D6(Vehicles::VehicleBogie& source, Vehicles::VehicleBogie& dest)
+    {
+        registers regs{};
+        regs.esi = X86Pointer(&source);
+        regs.edi = X86Pointer(&dest);
+        call(0x004AF4D6, regs);
+    }
+
+    static currency32_t vehicleRearrange(const VehicleRearrangeArgs& args, uint8_t flags)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/GameCommands/Vehicles/VehicleRearrange.cpp:30]: Could you add a comment with the function address?

File: src/OpenLoco/src/GameCommands/Vehicles/VehicleRearrange.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        }
+        else
+        {
+            auto* sourceBogie = sourceVehicle->asVehicleBogie();
+            if (sourceBogie == nullptr)
+            {
+                return FAILURE;
+            }
+            auto* destBogie = destVehicle->asVehicleBogie();
+            if (destBogie == nullptr)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/Vehicles/VehicleRearrange.cpp:88]: I had a thought about this last night. Destination could be just a head vehicle.  It's only source that will always be a bogie (car)

**Comment by duncanspumpkin** [comment]
> Rearranging train components seems to be working fine. Seems to be missing an error title, though.
> 
> Screenshot 2023-12-04 at 23 24 49

Interesting must be a mistake in the calling function

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**